doc/source/train/getting-started-xgboost.rst

```diff
@@ -0,0 +1,429 @@
+
+.. testcode:: python
+    :skipif: True
+
+    def train_func(config):
+        label_column = config["label_column"]
+        num_boost_round = config["num_boost_round"]
+        ...
+
+    config = {"label_column": "y", "num_boost_round": 10}
+    trainer = ray.train.xgboost.XGBoostTrainer(train_func, train_loop_config=config, ...)
+
+.. warning::
+
+    Avoid passing large data objects through `train_loop_config` to reduce the
+    serialization and deserialization overhead. Instead, it's preferred to
+    initialize large objects (e.g. datasets, models) directly in `train_func`.
+
+    .. code-block:: diff
+
+         def load_dataset():
+             # Return a large in-memory dataset
+             ...
+
+         def load_model():
+             # Return a large in-memory model instance
+             ...
+
+        -config = {"data": load_dataset(), "model": load_model()}
+
+         def train_func(config):
+        -    data = config["data"]
+        -    model = config["model"]
+
+        +    data = load_dataset()
+        +    model = load_model()
+             ...
+
+         trainer = ray.train.xgboost.XGBoostTrainer(train_func, train_loop_config=config, ...)
+
+Ray Train automatically sets up the Rabit communicator for XGBoost, which handles the distributed communication between workers.
+
+Report metrics and save checkpoints
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To persist your checkpoints and monitor training progress, add a
+:class:`ray.train.xgboost.RayTrainReportCallback` utility callback to your Trainer:
+
+
+.. code-block:: diff
+
+     import xgboost
+     from ray.train.xgboost import RayTrainReportCallback
+
+     def train_func():
+        ...
+        bst = xgboost.train(
+            ...,
+            callbacks=[
+                RayTrainReportCallback(
+                    metrics={"loss": "eval-logloss"}, frequency=1
+                )
+            ],
+        )
+        ...
+
+
+Reporting metrics and checkpoints to Ray Train enables integration with Ray Tune and :ref:`fault-tolerant training <train-fault-tolerance>`.
+
+Loading data
+------------
+
+When running distributed XGBoost training, you will need each worker to access a different shard of the dataset.
+
+
+.. testcode:: python
+    :skipif: True
+
+    def get_train_dataset(world_rank) -> xgboost.DMatrix:
+        # Define logic to get the DMatrix for each worker
+        ...
+
+    def get_eval_dataset(world_rank) -> xgboost.DMatrix:
+        # Define logic to get the DMatrix for each worker
+        ...
+
+    def train_func():
+        rank = ray.train.get_world_rank()
+        dtrain = get_train_dataset(rank)
+        deval = get_eval_dataset(rank)
+        ...
+
+A common way to do this is to pre-shard the dataset, so each worker reads from a different partition of the dataset. 
+
+For more flexibility, Ray Data provides a solution for sharding the dataset at runtime.
```

Comment:
```suggestion
A common way to do this is to pre-shard the dataset and then assign each worker a different set of files to read.

Pre-sharding the dataset is not very flexible to changes in the number of workers, since some workers may be assigned more data than others. For more flexibility, Ray Data provides a solution for sharding the dataset at runtime.
```